src/base/static/js/views/place-form-view.js

```diff
@@ -1,136 +1,259 @@
+
+      this.determineRenderabilityForEachCategory();
     },
-    resetFormState: function() {
-      this.formState = {
-        selectedCategoryConfig: {
-          fields: []
-        },
-        isSingleCategory: false,
-        attachmentData: null,
-        commonFormElements: this.options.placeConfig.common_form_elements || {}
-      }
-    },
-    render: function(isCategorySelected) {
-      var isAdmin = false,
-      self = this,
-      placesToIncludeOnForm = _.filter(this.placeDetail, function(place) { 
+
+    render: function() {
+      this.$el.html(Handlebars.templates['place-form']());
+      this.renderGeometryWarningMessage();
+      
+      var placesToIncludeOnForm = _.filter(this.placeDetail, function(place) { 
         return place.includeOnForm; 
       });
 
-      if (Shareabouts.bootstrapped.currentUser &&
-        _.contains(this.options.placeConfig.administrators, Shareabouts.bootstrapped.currentUser.username)) {
-        isAdmin = true;
-      }
-
-      // if there is only one place to include on form, skip category selection page
       if (placesToIncludeOnForm.length === 1) {
-        this.formState.isSingleCategory = true;
-        isCategorySelected = true;
+
+        // If we only have a single category, skip the category selection phase
         this.formState.selectedCategoryConfig = placesToIncludeOnForm[0];
+        this.renderFormFields();
+      } else {
+        this.renderCategoryButtons();      
       }
-      
-      this.checkAutocomplete();
 
+      return this;
+    },
+
+    renderGeometryWarningMessage: function() {
+      this.$el
+        .find(".place-form-geometry-messages")
+        .html(Handlebars.templates["place-form-messages"]());
+    },
+    
+    renderCategoryButtons: function() {
+      this.$el
+        .find(".place-form-category-buttons")
+        .html(Handlebars.templates["place-form-category-buttons"]({
+          placeConfig: this.options.placeConfig
+        }));
+    },
+
+    renderFormFields: function() {
       var data = _.extend({
-        isCategorySelected: isCategorySelected,
-        isAdmin: isAdmin,
         placeConfig: this.options.placeConfig,
         selectedCategoryConfig: this.formState.selectedCategoryConfig,
         user_token: this.options.userToken,
-        current_user: Shareabouts.currentUser,
-        isSingleCategory: this.formState.isSingleCategory
+        current_user: Shareabouts.currentUser
       }, Shareabouts.stickyFieldValues);
 
-      this.$el.html(Handlebars.templates['place-form'](data));
+      this.$el
+        .find("#place-form")
+        .html(Handlebars.templates["place-form-fields"](data));
 
-      if (this.center) $(".drag-marker-instructions").addClass("is-visuallyhidden");
-
-      $('#datetimepicker').datetimepicker({ formatTime: 'g:i a' });
+      if (this.geometryEnabled) {
+        this.options.appView.hideCenterPoint();
+        this.options.appView.hideSpotlightMask();
+        this.geometryEditorView.render({
```

Comment:
I wondered that myself. We reuse a single `GeometryEditorView` instance every time we need geometry editing capabilities, so we need to re-render it with different parameters depending on the context. I don't *think* there's an issue with this, but I'm not 100% sure about it either.